### PR TITLE
[lib/math-trig] Fix null undefined na parity with excel for math trig

### PR DESCRIPTION
**Pull request by pbadenski**
I simplified error checking to remove all or at least most of the duplication.

### CODE REVIEW

File: lib/math-trig.js [hunk 1]
```diff
@@ -116,6 +116,12 @@ exports.AGGREGATE = function(function_num, options, ref1, ref2) {
 };
 
 exports.ARABIC = function(text) {
+  if (text === undefined || text === null) {
```
Comment (pbadenski) [inline lib/math-trig.js:119]: From @nicolashefti (orig: https://github.com/formulajs/formulajs/pull/51/files#r659165805):

utils.parseNumber?

Comment (pbadenski) [reply lib/math-trig.js:119]: This check looks similar to the first 2 checks in parseNumber, but otherwise it isn't. Input to ARABIC is a roman numeral so it is text. Personally I'd leave it as is - as an exception.

File: lib/math-trig.js [hunk 2]
```diff
@@ -424,7 +470,11 @@ exports.FLOOR.PRECISE = exports.FLOOR.MATH;
 
 // adapted http://rosettacode.org/wiki/Greatest_common_divisor#JavaScript
 exports.GCD = function() {
-  var range = utils.parseNumberArray(utils.flatten(arguments));
+  var flatArguments = utils.flatten(arguments);
+  if (flatArguments.every(function (arg) { return arg === undefined || arg === null; })) {
```
Comment (pbadenski) [inline lib/math-trig.js:474]: From @nicolashefti (orig: https://github.com/formulajs/formulajs/pull/51/files#r659166060)

Can be checked in the existing loop

File: lib/math-trig.js [hunk 3]
```diff
@@ -602,7 +669,12 @@ exports.POWER = function(number, power) {
 };
 
 exports.PRODUCT = function() {
-  var args = utils.parseNumberArray(utils.flatten(arguments));
+  var flatArguments = utils.flatten(arguments);
+  var flatArgumentsDefined = flatArguments.filter(function (arg) { return arg !== undefined && arg !== null; });
```
Comment (pbadenski) [inline lib/math-trig.js:673]: FYI this can't be pulled into the loop as the behaviour of PRODUCT(undefined, 1) is different to PRODUCT(0, 1). parseNumber converts undefined to 1, so check for undefined needs to occur before conversion.

**Comment by pbadenski** [comment]
FYI:

 - first commit is the same as original changes (unless I made some silly mistake)
 - 980e125 and 6d21a4d - focus mostly on refactoring suggested in a previous pull request to simplify error handling
 - other commits are modifications to original pull request that I spotted during this rework. I tried to split them into smaller pieces of work and make commit messages explain the scope.
 - specifically I added a GT operator implementation which seemed to have been missing

File: lib/math-trig.js (hunk 2 above)
Comment (pbadenski) [reply lib/math-trig.js]: I removed this code in f827805

**Comment by nicolashefti** [comment]
@pbadenski that's the heavy one with math-trig.js :) I will start the review this weekend.

**Comment by pbadenski** [comment]
I understand that! :D Thank you.

**Comment by nicolashefti** [comment]
@pbadenski, no I don't mind :) thanks for the reminder. Summer, holidays, kids, ... too much good excuses. I'll work on that til the end of the week. Sorry about the delay.

File: lib/utils/common.js [hunk 4]
```diff
@@ -190,6 +207,16 @@ exports.parseDateArray = function(arr) {
   return arr;
 };
 
+exports.anyError = function() {
+  var n = arguments.length;
```
Comment (nicolashefti) [inline lib/utils/common.js:211]: Can you test this helper?

File: lib/utils/common.js [hunk 5]
```diff
@@ -200,6 +227,16 @@ exports.anyIsError = function() {
   return false;
 };
 
+exports.anyIsString = function() {
```
Comment (nicolashefti) [inline lib/utils/common.js:230]: Can you test this helper?

File: lib/math-trig.js (hunk 1 above)
Comment (nicolashefti) [reply lib/math-trig.js:119]: Good for me

**Review by nicolashefti — CHANGES_REQUESTED** [review]
@pbadenski thanks for the in-depth corner case research and fix. I like the new anyError helper. Can you just add a few test cases for the 2 new helpers?

File: lib/utils/common.js (hunk 5 above)
Comment (pbadenski) [reply lib/utils/common.js:229]: Done in e5d5967

File: lib/utils/common.js (hunk 4 above)
Comment (pbadenski) [reply lib/utils/common.js]: Done in e5d5967 - good catch, implementation was subtly wrong

**Review by nicolashefti — APPROVED** [review]
